addons/loyalty/models/loyalty_history.py

```diff
@@ -1,23 +1,41 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from datetime import datetime, timedelta
+
+from odoo import api, fields, models
+from odoo.tools import float_round
 
 
 class LoyaltyHistory(models.Model):
     _name = 'loyalty.history'
     _description = "History for Loyalty cards and Ewallets"
     _order = 'id desc'
 
-    card_id = fields.Many2one(comodel_name='loyalty.card', required=True, index=True, ondelete='cascade')
+    card_id = fields.Many2one(
+        comodel_name='loyalty.card',
+        required=True,
+        index=True,
+        ondelete='cascade',
+        readonly=True,
+    )
     company_id = fields.Many2one(related='card_id.company_id')
 
-    description = fields.Text(required=True)
+    description = fields.Text(required=True, readonly=True)
 
-    issued = fields.Float()
-    used = fields.Float()
+    available_issued_points = fields.Float()
+    issued = fields.Float(readonly=True)
+    used = fields.Float(readonly=True)
 
     order_model = fields.Char(readonly=True)
     order_id = fields.Many2oneReference(model_field='order_model', readonly=True)
+    is_line_expired = fields.Selection(
+        selection=[
+            ('expired', "Expired"),
+            ('valid', "Valid"),
+        ],
+        default='valid',
+    )
```

Comment:
Why is this needed to be stored instead of computed? Either the line don't have any expiry, or there is one that's before or after today. 
Quick look so it's not necessarily wrong, but I'd have imagined that searching for active points would have been based on that instead of a field depending on either a cron did pass before or not. :thinking: